designs/deprecated-ami-observability.md

```diff
@@ -0,0 +1,201 @@
+# Observability for Deprecated AMIs
+
+## Background
+
+With the recent introduction of a significant feature through [PR #6500](https://github.com/aws/karpenter-provider-aws/pull/6500) Karpenter has enhanced its capability to identify and utilize deprecated Amazon Machine Images (AMIs). Karpenter remains effective in provisioning new nodes within production environments where specific AMI IDs are mandated, adhering to these [guidelines](https://karpenter.sh/docs/tasks/managing-amis/#option-2-lock-down-which-amis-are-selected). Previously, if an AMI designated in an EC2NodeClass was deprecated, Karpenter faced challenges in launching new nodes, which could lead to potential service interruptions, especially in cases necessitating auto-scaling driven by Horizontal Pod Autoscalers (HPAs).
+
+This new feature would also benefit from enhanced observability, allowing cluster admins to identify which EC2NodeClasses are utilizing deprecated AMIs and take action accordingly.
+
+## Choices
+
+There are multiple possible solutions that can be implemented to add observability into usage of deprecated AMIs by EC2NodeClasses
+
+### Update the EC2NodeClass CRD
+
+This approach will modify the current CRD for the EC2NodeClass by adding a new `deprecated` field to the `status.amis` section, providing a clear and immediate indication of AMI deprecation directly within the resource configuration.
+
+#### Code Definition
+
+`pkg/apis/v1/ec2nodeclass_status.go`
+
+```go
+type AMI struct {
+    // ID of the AMI
+    // +required
+    ID string `json:"id"`
+    // Deprecation status of the AMI
+    // +optional
+    Deprecated bool `json:"deprecated"`
+    // Name of the AMI
+    // +optional
+    Name string `json:"name,omitempty"`
+    // Requirements of the AMI to be utilized on an instance type
+    // +required
+    Requirements []corev1.NodeSelectorRequirement `json:"requirements"`
+}
+```
+
+#### Proposed Spec
+
+``` yaml
+status:
+  amis:
+    - id: ami-01234567890654321
+      name: custom-ami-amd64
+      deprecated: true
+      requirements:
+        - key: kubernetes.io/arch
+          operator: In
+          values:
+            - amd64
+    - id: ami-01234567890123456
+      name: custom-ami-arm64
+      requirements:
+        - key: kubernetes.io/arch
+          operator: In
+          values:
+            - arm64
+```
+
+#### Pros + Cons
+
+* 👍 This will provide cluster admins a clear, concise indication of which AMIs being utilized by the EC2NodeClass are deprecated.
+* 👎 This will add a dependency to require a CRD update for the EC2NodeClass along with the version bump for Karpenter.
+
+### Add new status conditions to the EC2NodeClass CRD
+
+This is an alternate approach to update the status conditions for the EC2NodeClass to provide information to cluster admins that deprecated AMIs were discovered as part of the `amiSelectorTerms`.
```

Comment:
We should call out the nice behavior of getting additional metrics

Reply:
I have already mentioned the metric `operator_status_condition_count` as an example in the Pros and cons :)